File: desktop/src/protocol/personalization.ts
import { Type, type Static } from "@sinclair/typebox";

export const GlobalInstructionsSchema = Type.Object({
  content: Type.String(),
  path: Type.String(),
  overrideActive: Type.Boolean(),
});
export const MemorySettingsSchema = Type.Object({
  enabled: Type.Boolean(),
  allowExternalContext: Type.Boolean(),
});
export type GlobalInstructions = Static<typeof GlobalInstructionsSchema>;
export type MemorySettings = Static<typeof MemorySettingsSchema>;
export type MemorySettingsUpdate = Partial<MemorySettings>;

File: desktop/src-tauri/src/domain/personalization.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GlobalInstructions {
    pub content: String,
    pub path: String,
    pub override_active: bool,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct MemorySettings {
    pub enabled: bool,
    pub allow_external_context: bool,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct MemorySettingsUpdate {
    pub enabled: Option<bool>,
    pub allow_external_context: Option<bool>,
}
